macrosAutoCats/src/main/scala/com/softwaremill/macwire/autocats/internals/CatsProviders.scala

```diff
@@ -0,0 +1,147 @@
+package com.softwaremill.macwire.autocats.internals
+
+import scala.reflect.macros.blackbox
+import com.softwaremill.macwire.internals._
+import cats.implicits._
+
+trait CatsProviders[C <: blackbox.Context] {
+  val c: C
+  val log: Logger
+
+  val typeCheckUtil: TypeCheckUtil[c.type]
+
+  trait Provider {
```

Comment:
can this be sealed?

Reply:
Yes, It could be. I left it as a non-sealed trait because refactor required to introduce pure `autowire` is going to require adding new classes extending `Provider` trait. But for now, just to focus on the current use case, let's make it sealed.